message/router/middleware/poison_test.go

```diff
@@ -106,6 +111,53 @@ func TestPoisonQueue_handler_failing(t *testing.T) {
 	}
 }
 
+func TestPoisonQueue_context_values(t *testing.T) {
+	pubSub := gochannel.NewGoChannel(
+		gochannel.Config{Persistent: true},
+		watermill.NewStdLogger(true, true),
+	)
+
+	logger := watermill.NewStdLogger(true, true)
+
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
+	require.NoError(t, err)
+
+	pq, err := middleware.PoisonQueue(pubSub, "poison_queue")
+	require.NoError(t, err)
+	router.AddMiddleware(pq)
+
+	router.AddNoPublisherHandler("handler_name", "test", pubSub, func(msg *message.Message) error {
+		return errors.New("error")
+	})
+
+	go func() {
+		require.NoError(t, router.Run(context.Background()))
+	}()
+	require.NoError(t, err)
+	defer router.Close()
+
+	select {
+	case <-router.Running():
+	// ok
+	case <-time.After(time.Second):
+		t.Fatal("waiting for router timeout")
+	}
+
+	err = pubSub.Publish("test", message.NewMessage("1", nil))
+	require.NoError(t, err)
+
+	msgs, err := pubSub.Subscribe(context.Background(), "poison_queue")
+	require.NoError(t, err)
+
+	messages, all := subscriber.BulkRead(msgs, 1, time.Second)
+	require.True(t, all, "no messages received")
+
+	assert.Equal(t, "handler_name", messages[0].Metadata[middleware.PoisonedHandlerKey])
```

Comment:
those strings could be declared before creating the handler, and referenced by the variable name in both places.
Apart from `gochannel.Gochannel`, that comes from some more complicated logic

Reply:
> those strings could be declared before creating the handler, and referenced by the variable name in both places.
IMO when it's used twice using variable doesn't help a lot and obfuscates it a bit

> Apart from gochannel.Gochannel, that comes from some more complicated logic
yes, but here I'm not interested with this logic probably, but just with the outcome

Reply:
fair enough